run-checks

```diff
@@ -3,140 +3,194 @@
 export LANG=C.UTF-8
 export LANGUAGE=en
 
-if command -v goctest >/dev/null; then
-    goctest="goctest ./..."
-else
-    goctest="go test ./..."
-fi
+this_file="$0"
 COVERMODE=${COVERMODE:-atomic}
 COVERAGE_SUFFIX=${GO_BUILD_TAGS:-notags}
 COVERAGE_OUT=${COVERAGE_OUT:-.coverage/coverage-$COVERAGE_SUFFIX.cov}
 CHANGED_FILES=${CHANGED_FILES:-""}
 TIMEOUT=${TIMEOUT:-15}
+AUTO_INSTALL=${AUTO_INSTALL:-1}     # Automatically install supported tool dependencies (default: install)
+IGNORE_MISSING=${IGNORE_MISSING:-0} # Do not error on missing tool dependencies (default: error)
+
+tool_version() {
+    local tool=$1
+    case "$tool" in
+    go)
+        $tool version
+        ;;
+    ineffassign)
+        $tool -version 2>&1
+        ;;
+    misspell)
+        $tool -v
+        ;;
+    shellcheck)
+        $tool --version | grep version
+        ;;
+    golangci-lint) ;& # fallthrough
+    python3) ;&       # fallthrough
+    pytest-3) ;&      # fallthrough
+    clang-format)
+        $tool --version
+        ;;
+    nakedret)
+        echo "unknown"
+        ;;
+    *)
+        return 1
+        ;;
+    esac
+}
 
-if [ -z "${GITHUB_WORKFLOW:-}" ]; then
-    # when *not* running inside github, ensure we use go-1.18 by default
-    export PATH=/usr/lib/go-1.18/bin:"${PATH}"
-fi
-
-# add workaround for https://github.com/golang/go/issues/24449
-if [ "$(uname -m)" = "s390x" ]; then
-    if go version | grep -q go1.10; then
-        echo "covermode 'atomic' crashes on s390x with go1.10, reseting "
-        echo "to 'set'. see https://github.com/golang/go/issues/24449"
-        COVERMODE="set"
-    fi
-fi
-
-# If GOPATH is set in the shell environment, the path will be reflected
-# in $(go env GOPATH). If no shell path was set, Go will default the internal
-# GOPATH to $HOME/go. Note that GOPATH may contain a colon delimited set
-# of paths, so in order to run any binary from any of the installed GOPATHs
-# we must add all the possible bin locations.
-GOBINS=$(go env GOPATH | sed 's|:|/bin:|g' | sed 's|.*|\0/bin|g')
-export PATH="$PATH:$GOBINS"
-
-short=
-
-STATIC=
-UNIT=
-
-case "${1:-all}" in
-    all)
-        STATIC=1
-        UNIT=1
+tool_install() {
+    local tool=$1
+    case "$tool" in
+    ineffassign)
+        go install github.com/gordonklaus/ineffassign@latest >/dev/null
         ;;
-    --static)
-        STATIC=1
+    nakedret)
+        go install github.com/alexkohler/nakedret@v1.0.1 >/dev/null
         ;;
-    --unit)
-        UNIT=1
+    misspell)
+        go install github.com/client9/misspell/cmd/misspell@latest >/dev/null
         ;;
-    --short-unit)
-        UNIT=1
-        short=1
+    golangci-lint)
+        # Decided to start using latest instead of pinning specific version, if this approach
+        # introduces too much instability, revert back to pinning a working version.
+        go install github.com/golangci/golangci-lint/cmd/golangci-lint@latest >/dev/null
```

Comment:
turns out there's some instability due to v1.x -> v2.x switch see https://github.com/canonical/snapd/pull/15236 but our CI is using golangci-lint from snap which means we're already tracking the latest tag